app/libs/stats.py

```diff
@@ -1,33 +1,259 @@
+    months = months.group_by(fn.date_trunc('month', Comment.created))\
+        .order_by(SQL('date_trunc').asc())\
+        .tuples()
+    def output_formatter(mtc): return (
+        mtc[0].date().isoformat(), mtc[1]['name'], mtc[2])
+    for month in months:
+        monthly_top_commenters.extend(list(map(output_formatter,
+        Comment.select(fn.date_trunc('month', Comment.created), Comment.commenter, fn.count(Comment.id))\
+            .group_by(fn.date_trunc('month', Comment.created), Comment.commenter)\
+            .where(fn.date_trunc('month', Comment.created) == month)\
+            .order_by((SQL('date_trunc')), (SQL('count')).desc())\
+            .limit(int(top))\
+            .tuples())))
+    return monthly_top_commenters
+
+
+def monthly_top_commenters_lastNmonths(top=3, n=4):
+    return monthly_top_commenters(top, arrow.utcnow().shift(months=-int(n)).span('month')[0].date())
+ 
+
+def monthly_top_commented_articles(top=10, since=None):
+    monthly_top_commented_article = []
+    months = Comment.select(fn.date_trunc('month', Comment.created))
+    if since:
+        months = months.where(Comment.created >= since)
+    months = months.group_by(fn.date_trunc('month', Comment.created))\
+        .order_by(SQL('date_trunc').asc())\
+        .tuples()
+    def output_formatter(mtc): return (
+        mtc[0].date().isoformat(), Asset.get_by_id(mtc[1]).url, mtc[2])
+    for month in months:
+        monthly_top_commented_article.extend(list(map(output_formatter,
+        Comment.select(fn.date_trunc('month', Comment.created), Comment.asset_id, fn.count(Comment.id))\
+            .group_by(fn.date_trunc('month', Comment.created), Comment.asset_id)\
+            .where(fn.date_trunc('month', Comment.created) == month)\
+            .order_by((SQL('date_trunc')), (SQL('count')).desc())\
+            .limit(int(top))\
+            .tuples())))
+    return monthly_top_commented_article
+
+
+def monthly_top_commented_articles_lastNmonths(top=10, n=4):
+    return monthly_top_commented_articles(top, arrow.utcnow().shift(months=-int(n)).span('month')[0].date())
+
+
+def last2days_top_commented_articles(top=10):
+    last2days_top_commented_articles = Comment.select(Comment.asset_id, fn.count(Comment.id))\
+        .group_by(Comment.asset_id).where(Comment.created >= arrow.utcnow().shift(days=-int(2)).span('day')[0].date())\
+        .order_by((SQL('count')).desc())\
+        .limit(int(top))
+    return [(Asset.get_by_id(asset_id).url, count) for asset_id, count in last2days_top_commented_articles.tuples()]
+
 
-    data = zip(approved.tuples(), rejected.tuples())
-    return [(w1.date, c, (c + rc), round(c*100/(c+rc))) for (w1, c), (w2, rc) in data]
+# returns a dict containing total featured comments and their commenters with individual counts
+def featured_comments():
+    count = 0
+    featured_comments_commenter = {}
+    for commenter, is_featured_comment in Comment.select(Comment.commenter, Comment.editors_pick).tuples():
```

Comment:
@pramitrn you can apply where clause `Comment.editors_pick is True`
Also we can can group by query for count